### PR TITLE
Create and manage automations in the UI

### PR DESCRIPTION
**Pull request by pleek91**
# Description
Users can now manage their automations via the ui. Automations in Prefect enable you to configure [actions](https://docs.prefect.io/latest/concepts/automations/#actions) that Prefect executes automatically based on [trigger](https://docs.prefect.io/latest/concepts/automations/#triggers) conditions.

Read more about automations in the [prefect automation docs](https://docs.prefect.io/latest/concepts/automations/).

## Automation Triggers
The following preconfigured trigger templates are available
- Deployment status
- Flow run state
- Work pool status
- Work queue status

image

## Automation Actions
The following actions are available
- Cancel a flow run
- Suspend a flow run
- Change a flow run's state
- Pause a deployment
- Resume a deployment
- Pause a work queue
- Resume a work queue
- Pause a work pool
- Resume a work pool
- Pause an automation
- Resume an automation
- Send a notification

image

### CODE REVIEW

File: ui/src/components/AutomationTriggerTemplateSelect.vue [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+<script lang="ts" setup>
+  import { SelectOptionNormalized } from '@prefecthq/prefect-design'
+  import { AutomationTriggerTemplate, automationTriggerTemplates, getAutomationTriggerTemplateLabel } from '@prefecthq/prefect-ui-library'
+  import { computed } from 'vue'
+
+  const template = defineModel<AutomationTriggerTemplate | null>('template', { required: true })
+  /**
+   * Currently OSS doesn't have support for enabled/disabled trigger templates like cloud does.
+   * Only because it wasn't needed at the time of porting automations to OSS.
+   */
```
[the remark this answers is not in the document]
Comment (pleek91) [reply ui/src/components/AutomationTriggerTemplateSelect.vue:14]: ah the subtle difference between a comment and a docstring...

File: ui/src/pages/Automations.vue [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+  const subscription = useSubscription(api.automations.getAutomations)
+  const automations = computed(() => subscription.response ?? [])
+
+  const loaded = computed(() => subscription.executed)
+  const empty = computed(() => automations.value.length === 0)
+</script>
+
+<style>
+.automations\ {
```
[the remark this answers is not in the document]
Comment (pleek91) [reply ui/src/pages/Automations.vue:56]: 🙏

File: ui/src/components/AutomationWizardStepActions.vue [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+      addAction()
+    }
+  })
+
+  watch(actions, () => {
+    emit('update:automation', {
+      ...props.automation,
+      actions: actions.filter(isAutomationAction),
+    })
+  })
```
[the remark this answers is not in the document]
Comment (pleek91) [reply ui/src/components/AutomationWizardStepActions.vue:76]: So the local state `actions` can have partial/incomplete actions that won't pass validation. And failing validation prevents you from advancing in the wizard. But this acts as the `v-model` for actions on the automation itself. So it'll emit any valid/complete actions. 

Definitely not ideal but this is how its been working in cloud for a long time.

[the remark this answers is not in the document]
Comment (znicholasbrown) [reply ui/src/components/AutomationWizardStepActions.vue:76]: Ahh ok thanks for explaining, we can revisit this later as needed, seems ok for now

**Pull request merged**